model/trigger/task.go

```diff
@@ -363,3 +373,53 @@ func (t *taskTriggers) taskRegression(sub *event.Subscription) (*notification.No
 
 	return n, errors.Wrap(rec.Insert(), "failed to process regression trigger")
 }
+
+func (t *taskTriggers) taskExceedsTime(sub *event.Subscription) (*notification.Notification, error) {
+	if sub == nil || sub.TriggerData == nil {
+		return nil, fmt.Errorf("subscription %s is missing trigger data", sub.ID)
+	}
+	thresholdString, ok := sub.TriggerData[taskDurationKey]
+	if !ok {
+		return nil, fmt.Errorf("subscription %s has no task time threshold", sub.ID)
+	}
+	threshold, err := strconv.Atoi(thresholdString)
+	if err != nil {
+		return nil, fmt.Errorf("subscription %s has an invalid time threshold", sub.ID)
+	}
+	if t.event.EventType == event.TaskFinished {
+		maxDuration := time.Duration(threshold) * time.Second
+		if t.task.StartTime.Add(maxDuration).Before(t.task.FinishTime) {
+			return t.generate(sub, fmt.Sprintf("exceeded %d seconds", threshold))
+		}
+	}
+	return nil, nil
+}
+
+func (t *taskTriggers) taskRuntimeIncrease(sub *event.Subscription) (*notification.Notification, error) {
+	if sub == nil || sub.TriggerData == nil {
+		return nil, fmt.Errorf("subscription %s is missing trigger data", sub.ID)
+	}
+	percentString, ok := sub.TriggerData[percentIncreaseKey]
+	if !ok {
+		return nil, fmt.Errorf("subscription %s has no percentage increase", sub.ID)
+	}
+	percent, err := strconv.ParseFloat(percentString, 32)
+	if err != nil {
+		return nil, fmt.Errorf("subscription %s has an invalid percentage", sub.ID)
+	}
+
+	if t.event.EventType == event.TaskFinished {
+		lastGreen, err := t.task.PreviousCompletedTask(t.task.Project, []string{evergreen.TaskSucceeded})
```

Comment:
I believe this function can return `nil, nil`, so check that `lastGreen` is not nil.